src/service/repository/challenge.rs

```diff
@@ -140,4 +140,16 @@ impl Challenge {
                 FailedToGetChallenge(GetChallengeError(e)).into()
             })
     }
+
+    pub fn delete(connection: &mut PgConnection, challenge_id: &Uuid) -> Result<usize> {
+        use crate::schema::challenges::dsl::id;
+
+        match diesel::delete(challenges_table.filter(id.eq(challenge_id))).execute(connection) {
+            Ok(count) => Ok(count),
```

Comment:
No need to return anything in the Ok response since it's not being used anywhere
```suggestion
            Ok(_) => (),
```